pkg/kube/controllers/extendedjob/job_create_secret.go

```
@@ -0,0 +1,286 @@
+	for i := 0; i < len(pod.Spec.Containers)-1; i++ {
+		select {
+		case containerIndex := <-fileNotifyChannel:
+			outputContainer := pod.Spec.Containers[containerIndex]
+			err := createOutputSecret(outputContainer, namePrefix, namespace, pod.Name, clientSet, exjob)
+			if err != nil {
+				return err
+			}
+		case failure := <-errorChannel:
+			return errors.Wrapf(failure, "Failure waiting for output file for container in pod %s", pod.GetName())
+		}
+	}
+	return nil
+}
+
+// waitForFile waits for the file to be created
+func waitForFile(containerIndex int, filePath string, fileNotifyChannel chan<- int, errorChannel chan<- error) {
+
+	for {
+		time.Sleep(5 * time.Second)
+		_, err := os.Stat(filePath)
+		if err != nil {
+			if os.IsNotExist(err) {
+				continue
+			} else {
+				errorChannel <- err
+				break
+			}
+		}
+		fileNotifyChannel <- containerIndex
+		break
+	}
+}
+
+// authenticateInCluster authenticates with the in cluster and returns the client
+func authenticateInCluster() (*kubernetes.Clientset, *versioned.Clientset, error) {
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return nil, nil, errors.Wrapf(err, "failed to authenticate with incluster config")
+	}
+
+	clientSet, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, nil, errors.Wrapf(err, "failed to create clientset with incluster config")
+	}
+
+	versionedClientSet, err := versioned.NewForConfig(config)
+	if err != nil {
+		return nil, nil, errors.Wrapf(err, "failed to create versioned clientset with incluster config")
+	}
+
+	return clientSet, versionedClientSet, nil
+}
+
+func createOutputSecret(outputContainer corev1.Container, namePrefix string, namespace string, podName string, clientSet *kubernetes.Clientset, exjob *ejv1.ExtendedJob) error {
+
+	secretName := namePrefix + outputContainer.Name
+
+	// Fetch json from file
+	filePath := fmt.Sprintf("%s%s%s", "/mnt/quarks/", outputContainer.Name, "/output.json")
+	file, _ := ioutil.ReadFile(filePath)
```

❗️  you're ignoring an error here

---

Cared.